apps/src/code-studio/levels/saveAnswers.js

```diff
@@ -15,39 +15,30 @@ function saveAnswersAndNavigate(url) {
 
 const UNSUBMITTED_ATTEMPT = -50;
 
-// Currently just used by LevelGroup.js.
-// Called by gotoPage and checkForChanges to save current answers.
 // Calls the completeFn function when transmission is complete.
-// @param {function(function)} beforeHook Optional callback function to call
-//    before saveAnswers gets underway. This function must call the completion
-//    function passed as the first parameter to complete the saveAnswers call.
-// @param {Number} levelId Optional specific subLevelId that should be changed
 
-function saveAnswers(completeFn, beforeHook, subLevelId) {
-  if (beforeHook) {
-    beforeHook(sendResultsCompletion, subLevelId);
-  } else {
-    sendResultsCompletion();
-  }
-  function sendResultsCompletion() {
-    var results = window.getResult();
-    var response = results.response;
-    var result = results.result;
-    var submitted = appOptions.submitted || false;
+/**
+ * @param {function} completeFn - Function to call after sending report to
+ *   save answers.
+ */
+function saveAnswers(completeFn) {
```

Comment:
Get rid of `beforeHook`, and instead just call `beforeHook` before calling `saveAnswers`. Also get rid of `subLevelId` since it was only used by `beforeHook`